src/Jq/Expr.hs

```diff
@@ -63,6 +63,7 @@ data AbstractExpr n
     | Gt !Expr !Expr                 -- '>'
     | Leq !Expr !Expr                -- '<='
     | Geq !Expr !Expr                -- '>='
+    | AltDestr !Expr !Expr           -- '?//'
```

Comment:
I like that the name is abbreviated, rather than the full `AlternativeDestructure` that I suggested.

I would however like that:

2. `Destr` is expanded to `Destruct`, since `Destr` may be confused with "destroy".
0. `AltDestruct` is moved up to line 46 just below `Alternative`.
1. `Alternative` is renamed to `Alt`.

I will go ahead and rename `Alternative` to `Alt` once this PR is merged to relieve you.

```suggestion
    | AltDestruct !Expr !Expr           -- '?//'
```